File: app/src/main/java/ru/speinmerk/weather/room/dao/CityDao.java
package ru.speinmerk.weather.room.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import io.reactivex.Flowable;
import ru.speinmerk.weather.model.City;

@Dao
public interface CityDao {
    @Query("SELECT * FROM city")
    List<City> getAll();

    @Query("SELECT * FROM city")
    Flowable<List<City>> getAllListener();

    @Query("DELETE FROM city")
    void clearData();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(City city);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<City> cities);

    @Delete
    void delete(City city);

}

File: app/src/main/java/ru/speinmerk/weather/room/dao/CityDao.kt
package ru.speinmerk.weather.room.dao

import android.arch.persistence.room.*
import io.reactivex.Flowable
import ru.speinmerk.weather.model.City

@Dao
interface CityDao {
    @Query("SELECT * FROM city")
    fun getAll(): List<City>

    @Query("SELECT * FROM city")
    fun getAllListener(): Flowable<List<City>>

    @Query("DELETE FROM city")
    fun clearData()

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(city: City)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAll(cities: List<City>)

    @Delete
    fun delete(city: City)
}

